dali/operators/image/convolution/gaussian_blur_params.h

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_IMAGE_CONVOLUTION_GAUSSIAN_BLUR_PARAMS_H_
+#define DALI_OPERATORS_IMAGE_CONVOLUTION_GAUSSIAN_BLUR_PARAMS_H_
+
+#include <vector>
+
+#include "dali/core/dev_array.h"
+#include "dali/pipeline/data/views.h"
+#include "dali/pipeline/operator/common.h"
+
+namespace dali {
+namespace gaussian_blur {
+
+int SigmaToDiameter(float sigma) {
+  return 2 * ceilf(sigma * 3) + 1;
+}
+
+float DiameterToSigma(int diameter) {
+  // Based on OpenCV
+  int radius = (diameter - 1) / 2;
+  return (radius - 1) * 0.3 + 0.8;
+}
+
+struct DimDesc {
+  int usable_axes_start;
+  int usable_axes_count;
+  bool has_channels;
+  bool is_sequence;
+};
+
+template <int axes>
+struct GaussianBlurParams {
+  DeviceArray<int, axes> window_sizes;
+  DeviceArray<float, axes> sigmas;
+
+  bool IsUniform() const {
+    for (int i = 1; i < axes; i++) {
+      if (sigmas[i - 1] != sigmas[i] || window_sizes[i - 1] != window_sizes[i]) {
```

Comment:
I don't know how much it changes the perf but how about:
```
if (sigmas[0] != sigmas[i] || window_sizes[0] != window_sizes[i]) {
```

Reply:
I can change it in the next one, as the build will soon finish.